azure-pipelines.yml

```diff
@@ -64,4 +96,58 @@ jobs:
     inputs:
       command: custom
       verbose: false
-      customCommand: 'run compile'
\ No newline at end of file
+      customCommand: 'run compile'
+  - script: sudo npm install -g vsce
+    displayName: 'install vsce'
+  - script: node node_modules/gts/build/src/cli.js check
+    displayName: 'check gts error'
+    failOnStderr: True
+  - bash: |
+      is_prod_tag=^refs\/tags\/v?[0-9]+\.[0-9]+\.[0-9]+$
+      is_test_tag=^refs\/tags\/v?[0-9]+\.[0-9]+\.[0-9]+-[rR][cC]
+      [[ $BUILD_SOURCEBRANCH =~ $is_prod_tag ]] && is_prod=true && marketplace_token=$PROD_MARKETPLACE_TOKEN
+      [[ $BUILD_SOURCEBRANCH =~ $is_test_tag ]] && is_test=true && marketplace_token=$TEST_MARKETPLACE_TOKEN
+      if [[ "$BUILD_REASON" == "Schedule" ]]; then marketplace_token=$TEST_MARKETPLACE_TOKEN; fi
+      if [[ "$is_prod" == "true" || "$is_test" == "true" ]]; then deploy_to_github_release=true; fi
```

Comment:
Can we merge variable of deploy_to_github_release and deploy_to_marketplace to one variable? It seems they are the same value at any time.

In my option, the CI/CD flow would like this:
- For regular PR on develop branch, it has steps of 1) tslint; 2) complile; 3) run test
- For any bit ready to release (nightly build, RC, PROD release), it has extra steps of 4) build vsix package; 5) release to GitHub; 6) publish to marketplace;

we could use variable $need_release to differ regular PR from bit to release, the condition is like "tags start with version patter || build reason = schedule". And $need_release can be as additional condition of pipeline for steps of 4 ~ 6.

For regular PR, it use default package.json and don't need update config step, since it will not be release and only used for regular development.

For ready to release, it need update config to override configuration for release.

In this way, the valid job variable can be $need_release, $marketplace_token, which is only used in pipeline steps. And we could put $is_prod evaluated in updateConfig.js.

Reply:
Fixed.